op-service/sources/receipts_basic.go

```diff
@@ -46,6 +46,10 @@ func (f *BasicRPCReceiptsFetcher) FetchReceipts(ctx context.Context, block eth.B
 	if err != nil {
 		return nil, err
 	}
+	err = validateReceipts(block, bInfo.ReceiptHash(), txHashes, res)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This now leads to duplicated validation, because the RPCFetcher uses the BasicRPC fetcher

Reply:
That's right, but we also need the BasicRPC fetcher to do its own validation to prevent caching. If only the RPCFetcher does validation, the result would have gone through the caching layer too soon.